src/platform/packages/shared/content-management/table_list_view_table/src/table_list_view.test.tsx

```diff
@@ -330,6 +331,178 @@ describe('TableListView', () => {
     });
   });
 
+  describe('column sorting with custom columns', () => {
+    const customSortingOptions: CustomSortingOptions = {
+      field: 'typeTitle',
+      sortingLabels: [
+        {
+          label: 'Type A-Z',
+          direction: 'asc',
+        },
+        {
+          label: 'Type Z-A',
+          direction: 'desc',
+        },
+      ],
+    };
+    const customTableColumn = {
+      field: 'typeTitle',
+      name: 'Type',
+      sortable: true,
+    };
+
+    const setupCustomColumnSorting = registerTestBed<string, TableListViewTableProps>(
+      WithServices<TableListViewTableProps>(TableListViewTable, {
+        TagList: getTagList({ references: [] }),
+      }),
+      {
+        defaultProps: { ...requiredProps, customSortingOptions, customTableColumn },
+      }
+    );
+
+    const hits: Array<UserContentCommonSchema & { typeTitle: string }> = [
+      {
+        id: '123',
+        updatedAt: twoDaysAgo.toISOString(),
+        type: 'maps',
+        attributes: {
+          title: 'Item 2',
+        },
+        references: [],
+        typeTitle: 'Vega',
+      },
+      {
+        id: '456',
+        updatedAt: yesterday.toISOString(),
+        type: 'vega',
+        attributes: {
+          title: 'Item 1',
+        },
+        references: [],
+        typeTitle: 'Lens',
+      },
+    ];
+
+    test('should display 6 sorting options when custom columns are present', async () => {
+      let testBed: TestBed;
+
+      await act(async () => {
+        testBed = setupCustomColumnSorting({
+          findItems: jest.fn().mockResolvedValue({ total: hits.length, hits }),
+        });
+      });
+      const { openSortSelect } = getActions(testBed!);
+      const { component, find } = testBed!;
+      component.update();
+
+      act(() => {
+        openSortSelect();
+      });
+      component.update();
+
+      const filterOptions = find('sortSelect').find('li');
+
+      expect(filterOptions.length).toBe(6);
+      expect(filterOptions.map((wrapper) => wrapper.text())).toEqual([
+        'Name A-Z ',
+        'Name Z-A ',
+        'Type A-Z ',
+        'Type Z-A ',
+        'Recently updated. Checked option. ',
+        'Least recently updated ',
+      ]);
+      component.unmount();
+    });
+
+    test('should update the select option when custom column is selected', async () => {
+      const getCustomTableColumnSortButton = (testBed: TestBed, text: string) => {
```

Comment:
I realized that this method is using the column header for sorting. So our test coverage is currently missing a check that clicking on a custom sorting in the dropdown correctly updates the sorting. Ideally, we'd change this sorting interaction from column header to clicking the option from `filterOptions`